core/src/consensus/fork_choice.rs

```diff
@@ -54,7 +64,47 @@ pub trait ForkChoice {
     ) -> Vec<Self::ForkChoiceKey>;
 }
 
-fn select_forks_failed_switch_threshold(
+/*The idea here is that if we are waiting to switch due to FailedSwitchThreshold, reset_bank will be on the same fork as our last_voted_slot.
+We can then use the bank to figure out if our last vote has landed.
+
+This can be None if we have not voted before, or the fork from last_voted_slot is undergoing dump and repair due to duplicate consensus.
+For these cases we don't expect to be stuck long enough for super refresh to be needed so no reason to check the conditions. */
+
+fn last_vote_able_to_land(
+    reset_bank: Option<&Bank>,
+    progress: &ProgressMap,
+    tower: &Tower,
+) -> bool {
+    let Some(heaviest_bank_on_same_voted_fork) = reset_bank else {
+        // No reset bank means we are in the middle of dump & repair. Last vote
+        // landing is irrelevant.
+        return true;
+    };
+
+    let Some(last_voted_slot) = tower.last_voted_slot() else {
+        // No previous vote.
+        return true;
+    };
+
+    let Some(my_latest_landed_vote_slot) =
+        progress.my_latest_landed_vote(heaviest_bank_on_same_voted_fork.slot())
+    else {
```

Comment:
@AshwinSekar What would hitting this else case mean? We've never landed a vote or we've purged that fork?

Reply:
exactly we've either:
- never landed a vote (non voting validator) no reason to super refresh.
- fork has been pruned or is in the middle of dump & repair, so no need to super refresh.